docs/DeveloperGuide.md

```diff
@@ -1,33 +1,372 @@
+
+Note: A todo task has only task description while a deadline task has both the task description and a due date. 
+The due date must be in the pattern of ddMMyy as it is the pattern set by the `TimeParser` class.
+
+### Add an event item feature
+This feature is being facilitated by `AddCommand`.
+
+### Add a calendar item feature
+This feature is facilitated by `AddCommand`.
+The following sequence diagram shows how the `execute()` operation works:
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/add_command_SD.JPG" alt="" width="750"/><br/>
+
+Note: It checks the type of the new calendar item first, then it analyses the attached information, 
+and finally uses the corresponding adding function to add the new calendar item into the calendar list.  
+
+Note: It has seven command types, including `Todo`, `Deadline`, `Activity`, `Lecture`, `Tutorial`, `Lab` and ` Exam`.
+If the item is in `Lecture`, `Lab`, `Tutorial`, or `Exam` type, there will be a check in the function `isValid(command)` to ensure the module code included in the item is valid.
+If the item is in `Lecture`, `Lab`, or `Tutorial`type, it can be added as recurring items,
+while only one `Todo`, `Deadline`, `Exam` or `Activity` item can be added at a time.   
+
+### Mark a task as done feature
+This feature is facilitated by `DoneCommand`.
+The following sequence diagram show how the `execute()` operation works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/done_command_SD.JPG" alt="" width="750"/><br/>
+Note: It first extracts the task number from the user input prior to `convertTaskNumberToCalendarNumber`. 
+There will be a check in the function `markTaskAsDone(calendarNumber)` to ensure that the calendar item being marked as done is a task. 
+
+
+### Additional information of an event feature
+This feature is facilitated by `AddInfoCommand` and the `ViewInfoCommand`.
+The following sequence diagram show how the `execute()` operation of `AddInfoCommand` works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/addInfoCommand_SD.JPG" alt="" width="750"/><br/>
+The following sequence diagram show how the `execute()` operation of `ViewInfoCommand` works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/viewInfoCommand_SD.JPG" alt="" width="750"/><br/>
+
+Note: It first extracts the event number from the user input prior to `convertEventNumberToCalendarNumber`. 
+
+### Delete a calendar item feature
+This feature is facilitated by `DeleteCommand`.
+The following sequence diagram show how the `execute()` operation works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/deleteCommand_SD.JPG" alt="" width="750"/><br/>
+
+Note: It first extracts the task/event number from the user input prior to `convertTaskNumberToCalendarNumber` and `convertEventNumberToCalendarNumber` respectively. 
```

Comment:
There are light formatting errors in the diagrams here. Perhaps the life line needs to connect to the center of the object deletion cross. Some method calls are crossing the box boundaries.